### ISSUE TITLE
Inquiry about training data format

### ISSUE THREAD

**Issue by AlwaysTesseract** [lmarena/p2l#15]
Hi P2L team, thanks for the great work! Appreciate the paper and open-source code to go with it.

I am interested in using P2L for a domain specific use-case by training a custom P2L model. I was curious about the training data format required to do so. Specifically, Section 2 of the paper mentions a two-hot encoding:
> we let X represent a ‘two-hot’ encoding of the model pair, i.e., a vector of length M with zeros everywhere except +1 in the index B and −1 in the index A.

But the [p2l/dataset.py](https://github.com/lmarena/p2l/blob/d4c9cbd321d3c9f4dce2403c513e16282ac87f49/p2l/dataset.py#L9) module suggests a dataset with the following columns: `prompt`, `model_a`, `model_b`, `labels`. Further, the [p2l/model.py](https://github.com/lmarena/p2l/blob/d4c9cbd321d3c9f4dce2403c513e16282ac87f49/p2l/model.py#L101) module suggests that the `labels` column should have the form `(winner_idx, loser_idx)` for `bt` loss and `(winner_idx, loser_idx, tie_indicator)` for other loss functions, and no mentions of a two-hot encoding. 

Could you also clarify how the winner and loser index should be assigned in the case of ties?

Thanks in advance!

**Comment by efrick2002** [comment]
The formulation is the code is mathematically equivalent to the 2-hot proposed in the paper, while being a bit easier to work with w/r to bug-free coding, as well as easier to expand to the RK and GRK regression heads. The 2-hot is meant to index into the correct bt coefficients, in the code we just directly index into the relevant coefficients on the linear head.

In the case of a tie, the winner and loser index order doesn't matter–– either model can be the winner or loser.

Heres an example function I used to generate the labels:

```python
# models is a list containing model names, corresponding in position to the linear head.

def get_label_bb(row):
    winner = row['winner']

    if winner == 'model_a':
        winner_idx = models.index(row['model_a'])
        loser_idx = models.index(row['model_b'])
        tie_ind = 0
    elif winner == 'model_b':
        winner_idx = models.index(row['model_b'])
        loser_idx = models.index(row['model_a'])
        tie_ind = 0
    elif 'tie' == winner:
        winner_idx = models.index(row['model_a'])
        loser_idx = models.index(row['model_b'])
        tie_ind = 1
    elif 'tie (bothbad)' == winner:
        winner_idx = models.index(row['model_a'])
        loser_idx = models.index(row['model_b'])
        tie_ind = 2

    return [winner_idx, loser_idx, tie_ind]
```
